ktor-io/jvm/src/io/ktor/utils/io/Delimited.kt

```diff
@@ -52,11 +53,18 @@ suspend fun ByteReadChannel.skipDelimiter(delimiter: ByteBuffer) {
     }
 }
 
-private suspend fun ByteReadChannel.skipDelimiterSuspend(delimiter: ByteBuffer) {
+private suspend fun ByteReadChannel.skipDelimiterSuspend(delimiter: ByteBuffer): Boolean {
+    var result = true
+
     lookAheadSuspend {
-        awaitAtLeast(delimiter.remaining())
+        if (!awaitAtLeast(delimiter.remaining()) && !awaitAtLeast(1)) {
```

Comment:
this change is not obvious and requires test

Reply:
Reverted and added in the copied part (coverted by test)